oscar/cdk/stacks/lambda_stack.py

```diff
@@ -0,0 +1,260 @@
+from aws_cdk import (
+    Duration,
+    aws_lambda as lambda_,
+    aws_iam as iam,
+    aws_apigateway as apigateway,
+    aws_dynamodb as dynamodb,
+    CfnOutput
+)
+from constructs import Construct
+
+class OscarLambdaStack(Construct):
+    """
+    Lambda resources for OSCAR Slack Bot.
+    
+    This construct creates and configures the Lambda function, IAM role,
+    and API Gateway for the OSCAR Slack Bot.
+    """
+    
+    def __init__(
+        self, 
+        scope: Construct, 
+        construct_id: str, 
+        sessions_table: dynamodb.Table, 
+        context_table: dynamodb.Table
+    ) -> None:
+        """
+        Initialize Lambda resources.
+        
+        Args:
+            scope: The CDK construct scope
+            construct_id: The ID of the construct
+            sessions_table: The DynamoDB table for session data
+            context_table: The DynamoDB table for context data
+        """
+        super().__init__(scope, construct_id)
+        
+        # Create Lambda function role with appropriate permissions
+        self.lambda_role = self._create_lambda_role(sessions_table, context_table)
+
+        # Create Lambda function with placeholder code
+        self.lambda_function = self._create_lambda_function()
+
+        # Create API Gateway
+        self.api = self._create_api_gateway()
+        
+        # Add outputs for important resources
+        self._add_outputs()
+    
+    def _create_lambda_role(
+        self, 
+        sessions_table: dynamodb.Table, 
+        context_table: dynamodb.Table
+    ) -> iam.Role:
+        """
+        Create the IAM role for the Lambda function with appropriate permissions.
+        
+        Args:
+            sessions_table: The DynamoDB table for session data
+            context_table: The DynamoDB table for context data
+            
+        Returns:
+            The created IAM role
+        """
+        role = iam.Role(
+            self, "OscarLambdaRole",
+            assumed_by=iam.ServicePrincipal("lambda.amazonaws.com"),
+            managed_policies=[
+                iam.ManagedPolicy.from_aws_managed_policy_name("service-role/AWSLambdaBasicExecutionRole")
+            ]
+        )
+
+        # Add permissions for Bedrock
+        role.add_to_policy(
+            iam.PolicyStatement(
+                actions=[
+                    "bedrock:InvokeModel",
+                    "bedrock:RetrieveAndGenerate",
+                    "bedrock:Retrieve",
+                    "bedrock:GetFoundationModel",
+                    "bedrock:ListFoundationModels",
+                    "bedrock:GetKnowledgeBase",
+                    "bedrock:ListKnowledgeBases",
+                    "bedrock:GetInferenceProfile",
+                    "bedrock:ListInferenceProfiles",
+                    "bedrock-agent-runtime:Retrieve",
+                    "bedrock-agent-runtime:RetrieveAndGenerate",
+                    "bedrock-agent-runtime:InvokeAgent"
```

Comment:
Do we need all these permissions just by sending a prompt to bedrock?
In my experience with this we only need bedrock:InvokeModel.
https://github.com/opensearch-project/opensearch-metrics/blob/main/infrastructure/lib/stacks/gitHubAutomationApp.ts#L151-L160